identity/ctx_keys.go

```diff
@@ -0,0 +1,9 @@
+package identity
+
+// context key type and values
+type identityKey int
+
+const (
+	parsedKey identityKey = iota
```

Comment:
What is your reason for putting identityKey into a separate file? Isn't it basically another type?

Reply:
No other reason than breaking it apart a bit. Could be probably in the identity.go file as well, yeah. You want me to put it back?

Reply:
That would be my preference. I don't really like the pattern of separating types out into a file apart from methods or functions that use those types. I find it a lot easier to reason with if the type is as close to the methods or functions that use that type as possible, and a separate file just creates an arbitrary separation between them.

Reply:
Working on it.